components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/util/AccountUtil.java

```diff
@@ -145,4 +149,39 @@ public static boolean isTemplateExists(String templateType, String tenantDomain)
                             + "tenantDomain: " + tenantDomain, e);
         }
     }
+
+    /**
+     * Publishes an event.
+     *
+     * @param eventName                 Event name
+     * @param properties                Event properties
+     * @throws AccountLockException     if event sent failed
+     */
+    public static void publishEvent(String eventName, Map<String, Object> properties) throws AccountLockException {
+
+        Event identityMgtEvent = new Event(eventName, properties);
+        try {
+            AccountServiceDataHolder.getInstance().getIdentityEventService().handleEvent(identityMgtEvent);
+        } catch (IdentityEventException e) {
+            String errorMsg = "Error occurred while triggering the event : " + identityMgtEvent.getEventName();
+            throw new AccountLockException(errorMsg, e);
+        }
+    }
+
+    /**
+     * Clones the given map.
+     *
+     * @param map          Map.
+     * @return             Cloned Map.
+     */
+    public static Map<String, Object> cloneMap(Map<String, Object> map) {
+        if (map == null) {
```

Comment:
keep a new line after method declaration

Reply:
Fixed in 16e79d96ba3ce84d578113d59b183c512117ec94

Reply:
new line should be after `public static Map<String, Object> cloneMap(Map<String, Object> map) {` line

Reply:
Fixed in 5850c16fba894cba62d1f70e77d3088bc726777a

Reply:
Shall we use MapUtil.isEmpty() here?

Reply:
Fixed in 0203a22b19ea00df2e1ed6c9ad4f61421349e6eb